mempool/src/pool.rs

```diff
@@ -0,0 +1,330 @@
+use std::cmp::Ord;
+use std::cmp::Ordering;
+use std::collections::BTreeSet;
+use std::collections::HashMap;
+use std::collections::HashSet;
+use std::fmt::Debug;
+use std::rc::Rc;
+
+use parity_scale_codec::Encode;
+use thiserror::Error;
+
+use common::chain::transaction::Transaction;
+use common::chain::transaction::TxInput;
+use common::chain::OutPoint;
+use common::primitives::amount::Amount;
+use common::primitives::Id;
+use common::primitives::Idable;
+use common::primitives::H256;
+
+// TODO this willbe defined elsewhere (some of limits.rs file)
+const MAX_BLOCK_SIZE_BYTES: usize = 1_000_000;
```

Comment:
There's even an idea of having two different block sizes, one with gas and one with bytes. It's certainly gonna be more complex than that.

Reply:
OK. Fill me in when the time comes.

Reply:
Is there a specific reason all of these `bytes` constants are base 10 numbers and not nice base 2 numbers?

Reply:
Nope, arbitrary. Should I change them to base 2?

Reply:
It doesn't matter whether it's `1000000` or `2<<20`. To me it's more important for this to be configurable through a configuration object for p2p.